Code/Framework/AzToolsFramework/AzToolsFramework/ComponentModes/BoxComponentMode.cpp

```diff
@@ -357,4 +369,63 @@ namespace AzToolsFramework
         }
         return false;
     }
+
+    void InstallBaseShapeViewportEditFunctions(
+        BaseShapeViewportEdit* baseShapeViewportEdit, const AZ::EntityComponentIdPair& entityComponentIdPair)
+    {
+        auto getManipulatorSpace = [entityComponentIdPair]()
+        {
+            AZ::Transform manipulatorSpace = AZ::Transform::CreateIdentity();
+            ShapeManipulatorRequestBus::EventResult(
+                manipulatorSpace, entityComponentIdPair, &ShapeManipulatorRequestBus::Events::GetManipulatorSpace);
+            return manipulatorSpace;
+        };
+        auto getNonUniformScale = [entityComponentIdPair]()
+        {
+            AZ::Vector3 nonUniformScale = AZ::Vector3::CreateOne();
+            AZ::NonUniformScaleRequestBus::EventResult(
+                nonUniformScale, entityComponentIdPair.GetEntityId(), &AZ::NonUniformScaleRequestBus::Events::GetScale);
+            return nonUniformScale;
+        };
+        auto getTranslationOffset = [entityComponentIdPair]()
+        {
+            AZ::Vector3 translationOffset = AZ::Vector3::CreateZero();
+            ShapeManipulatorRequestBus::EventResult(
+                translationOffset, entityComponentIdPair, &ShapeManipulatorRequestBus::Events::GetTranslationOffset);
+            return translationOffset;
+        };
+        auto setTranslationOffset = [entityComponentIdPair](const AZ::Vector3& translationOffset)
+        {
+            ShapeManipulatorRequestBus::Event(
+                entityComponentIdPair, &ShapeManipulatorRequestBus::Events::SetTranslationOffset, translationOffset);
+        };
+        baseShapeViewportEdit->InstallGetManipulatorSpace(getManipulatorSpace);
+        baseShapeViewportEdit->InstallGetNonUniformScale(getNonUniformScale);
+        baseShapeViewportEdit->InstallGetTranslationOffset(getTranslationOffset);
+        baseShapeViewportEdit->InstallSetTranslationOffset(setTranslationOffset);
```

Comment:
suggestion: If using the value change mentioned earlier, we can `AZStd::move` these (or define them inline) and we'll save a copy as the values can be safely moved with a transfer of ownership.

Reply:
Thanks!